src/core/a-entity.js

```diff
@@ -461,22 +467,23 @@ var proto = Object.create(ANode.prototype, {
       var component = this.components[attr] || components[attr];
       var partialComponentData;
       value = value === undefined ? '' : value;
-      var valueStr = value;
+      var deserializedValue = value;
       var oldValue;
+      var serializedValue = value;
 
       if (component) {
         if (typeof value === 'string' && componentPropValue !== undefined) {
           // Update currently-defined component data with the new property value.
           partialComponentData = self.getAttribute(attr) || {};
           partialComponentData[value] = componentPropValue;
-          value = partialComponentData;
+          deserializedValue = partialComponentData;
```

Comment:
It's a common term but doesn't quickly parse in your brain. You have to think and it can be error prone if you are not 100% focused. Would you prefer JSON.serialize / JSON.deseralize instead of JSON.stringify / JSON.parse? What about componentString / componentObj or dataStr or dataObj?